src/main/java/dev/blaauwendraad/masker/json/SegmentReference.java

```diff
@@ -0,0 +1,20 @@
+package dev.blaauwendraad.masker.json;
+
+/**
+ * A mutable reference to a sequence of bytes in <code>message</code>. It is used to represent json path segments.
+ * A "key" segment type reference is represented as a (start, offset) pair.
+ * For an "array index" segment type reference, a (start, offset) pair is interpreted as (index, -1).
+ */
+public class SegmentReference {
+    int start;
+    int offset;
+
+    SegmentReference(int start, int offset) {
+        this.start = start;
+        this.offset = offset;
+    }
+
+    public boolean isArraySegment() {
```

Comment:
Maybe add some JavaDoc. Right now this is super clear to me, but I am afraid 1 year from now I might wonder what an `isArraySegment` means again

Reply:
Makes sense, I will do it.